### PR TITLE
Create users table

### PR DESCRIPTION
**Pull request by fonglh**
Insert a row into the users table on the user's first login.

### CODE REVIEW

File: web/controllers/auth_controller.ex [hunk 1]
```diff
@@ -32,6 +34,26 @@ defmodule Booky.AuthController do
...
     # Check if user belongs to the approved org
     if belong_to_github_org(client) do
+      # Check if user has already been added to the local database
+      unless Repo.get_by(User, username: user.username) do
+        # First login, insert into the db
+        changeset = User.changeset(%User{}, %{username: user.username, name: user.name})
+        case Repo.insert(changeset) do
+          {:ok, _} ->
+            conn
+            |> put_flash(:info, "Welcome to Booky!")
```
Comment (fonglh) [inline web/controllers/auth_controller.ex:44]: This and the 2 pipelines below only differ in the flash messages, but I can't figure out how to abstract it out.

The flash messages stop working if I try.

**Comment by fonglh** [comment]
Some other information needs to be stored, so I'll redo this in another PR.

1. Users cannot just be deleted when they are no longer with the organization as they may still have outstanding loans. They should be allowed to login, but can no longer borrow, and should be reminded to return any outstanding books.
1. An admin team is needed to manage the collection of books. This can be a GitHub organization team, but a flag needs to be created in the users table so the app doesn't have to keep checking the API.

**Comment by abeltay** [comment]
We should probably add a flag for `admin` and for `in organization` to handle 1 and 2 above.

This would mean that if an individual is out of the organization, as long as he still has outstanding loans, he will continue to have access. I suppose it shouldn't be a security issue. We can possibly restrict him from borrowing. What do you think?

What do you think about us manually adding a user (not in organization), just to keep track of books we may lend to other individuals?

**Comment by fonglh** [comment]
Yup I'm going to add those 2 flags, and allow login but not borrow for those who have left.

I'm ok with manually adding, but I think we can defer that.

**Pull request closed without merging**